android/app/build.gradle

```diff
@@ -184,50 +196,52 @@ android {
-                // https://issuetracker.google.com/issues/207403732
-                configureNdkBuildRelease.dependsOn(preReleaseBuild)
-                configureNdkBuildDebug.dependsOn(preDebugBuild)
-                project.ext.react.architectures.each { architecture ->
-                    tasks.findByName("configureNdkBuildDebug[${architecture}]")?.configure {
-                        dependsOn("preDebugBuild")
-                    }
-                    tasks.findByName("configureNdkBuildRelease[${architecture}]")?.configure {
-                        dependsOn("preReleaseBuild")
-                    }
-                }
-            } else if (reactNativeVersion > 0 && reactNativeVersion < 7100) {
-                // Due to a bug in AGP, we have to explicitly set a dependency
-                // between configureCMakeDebug* tasks and the preBuild tasks. This can
-                // be removed once this issue is resolved:
-                // https://issuetracker.google.com/issues/207403732
-                configureCMakeDebug.dependsOn(preDebugBuild)
-                configureCMakeRelWithDebInfo.dependsOn(preReleaseBuild)
-                project.ext.react.architectures.each { architecture ->
-                    tasks.findByName("configureCMakeDebug[${architecture}]")?.configure {
-                        dependsOn("preDebugBuild")
+            afterEvaluate {
+                preDebugBuild.dependsOn(packageReactNdkDebugLibs)
+                preReleaseBuild.dependsOn(packageReactNdkReleaseLibs)
+
+                if (reactNativeVersion < 7000) {
+                    // Due to a bug in AGP, we have to explicitly set a dependency
+                    // between configureNdkBuild* tasks and the preBuild tasks. This can
+                    // be removed once this issue is resolved:
+                    // https://issuetracker.google.com/issues/207403732
+                    configureNdkBuildRelease.dependsOn(preReleaseBuild)
+                    configureNdkBuildDebug.dependsOn(preDebugBuild)
+                    project.ext.react.architectures.each { architecture ->
+                        tasks.findByName("configureNdkBuildDebug[${architecture}]")?.configure {
+                            dependsOn("preDebugBuild")
+                        }
+                        tasks.findByName("configureNdkBuildRelease[${architecture}]")?.configure {
+                            dependsOn("preReleaseBuild")
+                        }
                     }
-                    tasks.findByName("configureCMakeRelWithDebInfo[${architecture}]")?.configure {
-                        dependsOn("preReleaseBuild")
+                } else {
+                    // Due to a bug in AGP, we have to explicitly set a dependency
+                    // between configureCMakeDebug* tasks and the preBuild tasks. This can
+                    // be removed once this issue is resolved:
+                    // https://issuetracker.google.com/issues/207403732
```

Comment:
this seems to be fixed in AGP 7.3.0-alpha9; since 0.71 uses AGP 7.3.1 should we have a separate if statement for 7100?

Reply:
Starting with 7100, `usePrefabs` is true, so we won't go into this path any longer. Eventually, we will just remove this branch.